core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -242,6 +242,53 @@ public void transportServiceAdapter(TransportServiceAdapter service) {
         this.transportServiceAdapter = service;
     }
 
+    private static class VersionHandshakeResponseTransportResponseHandler<Channel> implements
```

Comment:
Since it's a private class, maybe a shorter name like `VersionHandshakeHandler` or `HandshakeResponseHandler` is fine?